src/main/java/org/folio/linked/data/service/impl/ResourceServiceImpl.java

```diff
@@ -257,4 +259,23 @@ private void breakEdgesAndDelete(Resource resource) {
     breakCircularEdges(resource);
     resourceRepo.delete(resource);
   }
+
+  private void refreshWork(Resource resource) {
+    if (resource.isOfType(INSTANCE)) {
+      extractWorkFromInstance(resource)
+        .ifPresent(work -> {
+          edgeRepo.findByIdTargetHash(work.getId())
+            .forEach(work::addIncomingEdge);
+          addOutgoingEdges(work);
```

Comment:
Hi @askhat-abishev, What is the need to call addOutgoingEdges from here?

Reply:
Hi @pkjacob, It is required as not all the necessary data is fetched from the database (e.g. classification)